src/stream/src/executor/hash_join.rs

```diff
@@ -725,27 +742,27 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
     async fn hash_eq_match(
         key: &K,
         ht: &mut JoinHashMap<K, S>,
-    ) -> StreamExecutorResult<Option<HashValueType>> {
-        if !key.null_bitmap().is_subset(ht.null_matched()) {
-            Ok(None)
+    ) -> StreamExecutorResult<CacheResult> {
+        if enable_strict_consistency() {
+            Ok(ht.take_state_opt(key))
         } else {
-            ht.take_state(key).await.map(Some)
+            ht.take_state(key).await.map(CacheResult::Hit)
```

Comment:
If we enable strict consistency, we can avoid having to fetch all state into cache.

However if we tolerate inconsistency, we must fetch all state into cache before handling the matching. Otherwise we can't normalize the inconsistent state between degree table and corresponding state table.